mantidproject/mantid
# Masking workspaces with grouped spectra

fixes #16808 which may be considered a bug or very peculiar feature as the code provided below does not work until you apply this ticket changes. See detailed description of the issue within the ticket. 

To allow masking source workspace with mask, obtained from reduced workspace (which usually have different spectra numbers) I had to change a bit the algorithm logic which is reflected in unit tests and modified algorithm description. 

**To test:**

Behaviour have changed and Documentation reflects that.  System tests should assure nothing wrong happened. I made some changes to system tests to address the bugs, which become apparent while making the changes. 

Load ISIS run or any run with grouped detectors or non-trivial spectra-detector map (e.g. ISIS MAP-s file from system tests would be good), open instrument, view Draw a mask, apply it to data and save it to xml file:
e.g.:

``` Python
Load(Filename=r'.../Datastore/DataCopies/Testing/Data/SystemTest/MAP17186.raw', OutputWorkspace='MAP17186')
# drowing mask on instrument have occurred here
# here are commands which extract this mask
ExtractMask(InputWorkspace='MAP17186', OutputWorkspace='__InstrumentActor_MaskWorkspace')
MaskDetectors(Workspace='MAP17186', MaskedWorkspace='__TMP0000021CE2A7F2D0')

#Save the mask from 'MAP17186' workspace to xml file using SaveMask algorithm
SaveMask(InputWorkspace='MAP17186',OutputFIle='MapsMask1.xml')
```

Load another MAPS workspace

``` Python
Load(Filename=r'.../Datastore/DataCopies/Testing/Data/SystemTest/MAP17269.raw', OutputWorkspace='MAP17269')

#Load saved mask
LoadMask(Instrument='MAP', InputFile='D:/Data/Mantid_Testing/2016/MapsMask1.xml',OutputWorkspace='Masking')
```

Note that number of spectra in the loaded mask is different from the the one loaded.

Mask the loaded workspace:
`MaskDetectors(Workspace='MAP17269', MaskedWorkspace='Masking')`

Before the changes, this would fail. After the changes, on instrument view you would see exactly the same masked area as on the initial masked workspace. 

Release notes modified accordingly. 

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
